src/ralph/models/edx/textbook_interaction.py

```diff
@@ -0,0 +1,627 @@
+        caseSensitive (bool): Consists either of the `true` value if the case sensitive
+            option is selected or `false` if this option is not selected.
+        highlightAll (bool): Consists either of the `true` value if the option to highlight
+            all matches is selected or `false` if this option is not selected.
+        query (str): Consists of the value in the search field.
+        status (str): Consists either of the value `not found` for a search string that is
+            unsuccessful or blank for successful search strings.
+    """
+
+    name: Literal["textbook.pdf.search.highlight.toggled"]
+    caseSensitive: bool
+    highlightAll: bool
+    query: str
+    status: Union[Literal[""], Literal["not found"]]
+
+
+class TextbookPdfSearchCaseSensitivityToggledEventField(
+    TextbookInteractionBaseEventField
+):
+    """Representents the `textbook.pdf.searchcasesensitivity.toggled` event field.
+
+    Attributes:
+        name (str): Consists of the value `textbook.pdf.searchcasesensitivity.toggled`.
+        caseSensitive (bool): Consists either of the `true` value if the case sensitive
+            option is selected or `false` if this option is not selected.
+        highlightAll (bool): Consists either of the `true` value if the option to highlight
+            all matches is selected or `false` if this option is not selected.
+        query (str): Consists of the value in the search field.
+        status (str): Consists either of the value `not found` for a search string that is
+            unsuccessful or blank for successful search strings.
+    """
+
+    name: Literal["textbook.pdf.searchcasesensitivity.toggled"]
+    caseSensitive: bool
+    highlightAll: bool
+    query: str
+    status: Union[Literal[""], Literal["not found"]]
+
+
+class BookEventField(AbstractBaseEventField):
+    """Representents the `book` event field.
+
+    Attributes:
+        chapter (str): Consists of the name of the PDF file.
+        name (str): Consists of `textbook.pdf.page.loaded` if type is set to `gotopage`,
+            `textbook.pdf.page.navigatedprevious` if type is set to `prevpage`,
+            `textbook.pdf.page.navigatednext` if type is set to `nextpage`.
+        new (int): Consists of the destination page number.
+        old (int): Consists of the original page number. It applies to `gotopage` event types only.
+        type (str): Consists of `gotopage` value when a page loads after the student manually
+            enters its number, `prevpage` value when the next page button is clicked or `nextpage`
+            value when the previous page button is clicked.
+    """
+
+    chapter: constr(
+        regex=(
+            r"^block-v1:[^\/+]+(\/|\+)[^\/+]+(\/|\+)[^\/?]+type"  # noqa : F722
+            r"@sequential\+block@[a-f0-9]{32}.+\.pdf$"  # noqa : F722
+        )
+    )
+    name: Union[
+        Literal["textbook.pdf.page.loaded"],
+        Literal["textbook.pdf.page.navigatedprevious"],
```

Comment:
We should rather believe it rather than the [documentation](https://edx.readthedocs.io/projects/devdata/en/stable/internal_data_formats/tracking_logs.html#book). Upon reflection, `pdf.navigatedprevious` event does not exist, hence, it would be a non sense